src/Streamly/Internal/Data/Array/Foreign/Type.hs

```diff
@@ -139,7 +138,11 @@ data Array a =
     Storable a =>
 #endif
     Array
-    { arrContents :: {-# UNPACK #-} !ForeignPtrContents -- first address
+    { arrContents ::
+#ifndef USE_FOREIGN_PTR
```

Comment:
Don't you need to initialize the flag accordingly here?

Reply:
Remains undefined by default.